packages/ui/src/components/DatePicker/SDatePicker.vue

```diff
@@ -0,0 +1,734 @@
+  const newVal = event.target.value
+  const date = fromFormat(newVal)
+  if (date) (rangeState as any).value[field as keyof RangeState] = date
+  rangeState.value.selecting = false
+  updateModelValue()
+  updateShowedMonths()
+}
+
+const updateCustomInputDay = (event: any) => {
+  const newVal = event.target.value
+  const date = fromFormat(newVal)
+  if (date) dayState.value = date
+  updateModelValue()
+  updateShowedMonths()
+}
+
+const updateCustomInputPick = (event: any) => {
+  const newVal = event.target.value
+  const date = fromFormat(newVal)
+  if (date) {
+    if (pickState.value.length > 0) pickState.value[pickState.value.length - 1] = date
+    else pickState.value.push(date)
+  }
+  updateModelValue()
+  updateShowedMonths()
+}
+
+const customInputValuePick = computed(() => {
+  return formatDate(pickState.value[pickState.value.length - 1])
+})
+
+const customInputValueStartDate = computed(() => {
+  return customInputValue('startDate')
+})
+
+const customInputValueEndDate = computed(() => {
+  return customInputValue('endDate')
+})
+
+const customInputValueDay = computed(() => {
+  return formatDate(dayState.value)
+})
+
+const customInputCssClass = computed(() => {
+  return `custom-panel__input${props.time ? '_time' : ''}`
+})
+
+function mask(this: any, event: any) {
+  const date = this.value
+  if (event.inputType === 'deleteContentBackward') {
+    return
+  } else if (date.match(/^\d{2}$/) !== null) {
+    this.value = date + '/'
+  } else if (date.match(/^\d{2}\/\d{2}$/) !== null) {
+    this.value = date + '/'
+  } else if (date.match(/^\d{2}\/\d{2}\/\d{4}$/) !== null) {
+    if (props.time) this.value = date + ', '
+  } else if (date.match(/^\d{2}\/\d{2}\/\d{4}\, \d{2}$/) !== null) {
+    this.value = date + ':'
+  } else if (date.match(/^\d{2}\/\d{2}\/\d{4}\, \d{2}\:\d{2}$/) !== null) {
+    this.value = date
+  }
+}
+
+const customInputLength = computed(() => {
+  return props.time ? 17 : 10
+})
+
+// #endregion
+
+// #region FORMATTING
+
+const formatPattern = computed(() => {
+  return props.time ? 'dd/MM/yyyy, HH:mm' : 'dd/MM/yyyy'
+})
+
+const formatDate = (date: any) => {
+  if (!date) return ''
+  // return DateTime.fromJSDate(date).toFormat(formatPattern.value)
+  return format(date, formatPattern.value)
+}
+
+const fromFormat = (dateString: string) => {
+  // let date = DateTime.fromFormat(dateString, formatPattern.value)
+  let date = parse(dateString, formatPattern.value, new Date())
+  if (!isValid(date)) return null
+  console.dir(date)
+  return date
+}
+
+// #endregion
+
+// #region POPPER
+const [showPopper, togglePopper] = useToggle(false)
+whenever(and(props.disabled, showPopper), () => togglePopper(false), { immediate: true })
+
+const updateShow = () => {
+  if (!props.disabled) togglePopper(true)
+}
+// #endregion
+
+if (props.modelValue || (props.modelValue as Date[])?.length > 0) init()
+else updateModelValue()
+</script>
+
+<template>
+  <div class="date-picker-component">
```

Comment:
### General note about classes naming and BEM

- The only one BEM element should be used within the component.
- We always use `s-` prefix for all BEM in the library: `s-modal`, `s-text-field` etc

Thus, I ask you to use only `s-date-picker` in this component. Also I see that you use `date-picker` and `date-picker-component` - these names seems inconsistent.